pym/portage/package/ebuild/config.py

```diff
@@ -2729,6 +2729,14 @@ def environ(self):
 		if not eapi_exports_merge_type(eapi):
 			mydict.pop("MERGE_TYPE", None)
 
+		if eapi_attrs.sysroot and _phase_func_map.get(phase, '').startswith('src_'):
+			# TODO: implement Portage-wide support for custom SYSROOT
+			mydict["SYSROOT"] = (
+					self.get('SYSROOT', '').rstrip(os.path.sep) + os.path.sep)
+			mydict["ESYSROOT"] = (
+					mydict["SYSROOT"].rstrip(os.path.sep) + self.get("EPREFIX", "")
+					+ os.path.sep)
```

Comment:
It feels awkward having a trailing slash here (since we have just removed if from ${D} and ${ED}). If we really need the trailing slash, if would we nice to have an explanation somewhere.

Reply:
Okay, good.